### PR TITLE
Implement ignoring auto-fixable codes

### LINKED ISSUE #65: Automatically ignore some whitespace errors
For good reasons, settings like `max-line-length` and `max-complexity` should be not held within sublime settings, but in `.flake8` or another config file. 

The exception here should be the typical whitespace warnings Sublime will autocorrect for us on save. These are:

W291 | trailing whitespace
-- | --

W293 | blank line contains whitespace
-- | --

W391 | blank line at end of file
-- | --

### PR DESCRIPTION
**Pull request by kaste**
Fixes #65

### CODE REVIEW

**Comment by braver** [comment]
Do add something to the readme about this. The default should also be false, fixing white space isn’t on by default in Sublime Text.

File: linter.py [hunk 1]
```diff
@@ -8,7 +8,9 @@ class Flake8(PythonLinter):
 
     cmd = ('flake8', '--format', 'default', '${args}', '-')
     defaults = {
-        'selector': 'source.python'
+        'selector': 'source.python',
+        # By default, filter codes Sublime can auto-fix
+        'ignore-fixables': True
```
Comment (braver) [inline linter.py]: - default: False
- document how to enable
- document which codes it ignores
- document which settings to enable in ST

File: linter.py [hunk 2]
```diff
@@ -38,6 +40,33 @@ class Flake8(PythonLinter):
...
+        settings = self.get_view_settings()
+        errors = super().parse_output(proc, virtual_view)
+
+        if not settings.get('ignore-fixables', False):
+            return errors
+
+        filtered_errors = []
+        for error in errors:
+            code = error['code']
+            if code in ('W291', 'W293'):  # no 'trailing' WS errors
```
Comment (braver) [inline linter.py]: Don't forget W292: "no newline at end of file"

**Comment by kaste** [comment]
Hm, probably better to first just check which 'fixers' are actually enabled. Added W292 support.

**Comment by braver** [comment]
Nifty!

**Comment by kaste** [comment]
BTW still think `ignore_fixables` should be False by default now that I actually check which of the fixers are on?

**Comment by braver** [comment]
On the fence really. Let’s go with True and see what happens.

**Comment by kaste** [comment]
Ok, this is now release candidate 4.2.0.

**Comment by kaste** [comment]
Erm, same code just added the documentation. Code is 3 months old, so it probably works.

**Comment by kaste** [comment]
You know, we don't need to rush through it

**Comment by braver** [comment]
I’m gonna play with this tomorrow, but am away next week (in the German forests far away from my laptop) so, I guess I’ll merge it when I get back?

**Comment by braver** [comment]
Flake8 is broken for me right now, so can't test: https://gitlab.com/pycqa/flake8/issues/437

**Comment by kaste** [comment]
That's a soft erred bc flake8 puts something on stderr if you use py37. The linter will still work. Anyway, fixed it!

We discussed this before, and thought flake will probably do a release as soon as py37 gets a final release. Nope, of course.

**Comment by braver** [comment]
Sweet!

**Comment by braver** [comment]
Ok, so I didn't give it a big workout, but it's really nice. Suppressing the FutureWarning stuff is super nice. Let's 🛥

**Pull request merged**